src/Adapter/MSTest.CoreAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -129,5 +136,26 @@ internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElem
                 PlatformServiceProvider.Instance.FileOperations.DisposeNavigationSession(navigationSession);
             }
         }
+
+        private bool CollectSourceInformation(IRunSettings runSettings)
```

Comment:
This logic should be part of the [MSTestSettings](https://github.com/Microsoft/testfx/blob/master/src/Adapter/MSTest.CoreAdapter/MSTestSettings.cs) class which parses the runsettings already.